src/Extensions/Annotatable.php

```diff
@@ -8,68 +14,69 @@
  *
  * @package IDEAnnotator/Extensions
  *
- * @property DevBuildController $owner
+ * @property \SilverStripe\Dev\DevBuildController $owner
  */
 class Annotatable extends Extension
 {
 
     /**
-     * @var DataObjectAnnotator
-     */
-    protected $annotator;
-
-    /**
-     * @var AnnotatePermissionChecker
-     */
-    protected $permissionChecker;
-
-    /**
      * Keep track ot the annotation actions for extensions
      * An Extension can belong to many DataObjects.
      * This prevents that an Extension is ran twice on dev/build
      * @var array
      */
     public static $annotated_extensions = array();
-
     /**
-     * Annotatable setup.
-     * This is theoratically a constructor, but to save memory we're using setup called from {@see requireDefaultRecords}
+     * @var DataObjectAnnotator
      */
-    public function setUp()
-    {
-        $this->annotator = Injector::inst()->get('DataObjectAnnotator');
-        $this->permissionChecker = Injector::inst()->get('AnnotatePermissionChecker');
-    }
+    protected $annotator;
+    /**
+     * @var AnnotatePermissionChecker
+     */
+    protected $permissionChecker;
 
     /**
      * Annotated Controllers and Extensions
+     * @throws \Psr\Container\NotFoundExceptionInterface
      */
     public function afterCallActionHandler()
     {
         $this->setUp();
 
         $skipAnnotation = $this->owner->getRequest()->getVar('skipannotation');
-        $envIsAllowed   = $this->permissionChecker->environmentIsAllowed();
-        
+        $envIsAllowed = Director::get_environment_type() === 'dev';
+
         if ($skipAnnotation === null && $envIsAllowed) {
 
-            $this->displayMessage(' Generating class docblocks');
+            // @todo make this work with a proper list <ul>. Or CLI equivalent.
+            $this->displayMessage("<div class='build'><p><b>Generating class docblocks</b></p><ul>\n\n");
 
             $modules = $this->permissionChecker->enabledModules();
             foreach ($modules as $module) {
                 $this->annotator->annotateModule($module);
             }
 
-            $this->displayMessage(' Docblock generation finished!');
+            $this->displayMessage("</ul>\n<p><b>Docblock generation finished!</b></p></div>");
         }
     }
 
     /**
+     * Annotatable setup.
+     * This is theoratically a constructor, but to save memory we're using setup called from {@see requireDefaultRecords}
+     * @throws \Psr\Container\NotFoundExceptionInterface
+     */
+    public function setUp()
+    {
+        $this->annotator = Injector::inst()->get(DataObjectAnnotator::class);
+        $this->permissionChecker = Injector::inst()->get(AnnotatePermissionChecker::class);
```

Comment:
You can use injector dependencies for this if you wanted to:

```php
private static $dependencies = [
    'annotator' => '%$' . DataObjectAnnotator::class,
    'permissionChecker' => '%$' . AnnotatePermissionChecker::class,
];

public function getAnnotator();
public function setAnnotator(DataObjectAnnotator $annotator);
public function getPermissionChecker();
public function setPermissionChecker(AnnotatePermissionChecker $permissionChecker);
```

Reply:
Rather not, because it would do it on construction, which is exactly what we tried to avoid by not putting it in an actual constructor in the first place